tasks/determine-image-expiration-task.yaml

```
@@ -33,7 +33,8 @@ spec:
       echo "Target branch: ${PIPELINE_TARGET_BRANCH}"
 
       # TODO(ROX-28001): Update the expiration logic for -fast releases.
-      if [[ "${PIPELINE_EVENT_TYPE}" == "push" && "${PIPELINE_TARGET_BRANCH}" == refs/tags/* ]]; then
+      if [[ "${PIPELINE_EVENT_TYPE}" != "pull_request" && "${PIPELINE_TARGET_BRANCH}" == refs/tags/* && "${PIPELINE_TARGET_BRANCH}" != refs/tags/*-nightly-* ]]; then
```

[nit] There's no need to mention `refs/tags/` in the last comparison:

```suggestion
      if [[ "${PIPELINE_EVENT_TYPE}" != "pull_request" && "${PIPELINE_TARGET_BRANCH}" == refs/tags/* && "${PIPELINE_TARGET_BRANCH}" != *-nightly-* ]]; then
```

---

Done in https://github.com/stackrox/konflux-tasks/pull/37/commits/0ddbd45cfd37e680ac1b1ad4082aac6447784acc